gmso/core/virtual_site.py

```diff
@@ -68,10 +85,23 @@ def position(self) -> str:
             raise MissingPotentialError(
                 "No VirtualPositionType associated with this VirtualType."
             )
-        # TODO: validate parent atoms matches virtual_type.virtual_position in terms of independent variables ri, rj, etc.
-        # TODO: Generate position from atoms of parent_atoms and self.virtual_type.virtual_position.expression.
-        raise NotYetImplementedWarning(
-            "Need a functional to call from self.virtual_type.virtual_position, and plug in ri, rj, rk etc."
+
+        independent_namespace = {}
+        for _, symbol in zip(range(len(self.parent_sites)), string.ascii_lowercase[8:]):
+            x, y, z = symbols(f"r{symbol}1 r{symbol}2 r{symbol}3")
+            independent_namespace[f"r{symbol}"] = Matrix([x, y, z])
+
+        independent_parameters = {}
+        for symbol, site in zip(string.ascii_lowercase[8:], self.parent_sites):
+            for i, pos in enumerate(site.position):
+                independent_parameters[f"r{symbol}{i + 1}"] = float(pos.value)
+
+        # perform expression evaluation
+        return (
+            self.virtual_type.virtual_position.potential_expression.evaluate(
+                independent_namespace, independent_parameters
+            ).T
+            * u.nm
```

Comment:
We should get the units from the parent site positions and use that here instead of hard-coding nanometers. Or maybe in this case it would come from the units in the virtual site type.

Reply:
Completed in 7f936662